connectivity/tests/encryption.go

```diff
@@ -56,14 +56,16 @@ func getInterNodeIface(ctx context.Context, t *check.Test, clientHost *check.Pod
 	return device
 }
 
-// getSourceAddress determines the source IP address we want to use for
-// capturing packet. If direct routing is used, the source IP is the client IP.
-func getSourceAddress(ctx context.Context, t *check.Test, client,
+// getSourceAddressFilter constructs the source IP address filter we want to use for
+// capturing packet. If direct routing is used, the source IP is the client IP,
+// otherwise, either the client IP or the one it can be masqueraded to.
```

Comment:
Nit: I think in the node-to-node encryption w/ tunneling on case the IP addr is just cilium_host IP addr, and it is not masqueraded to.

Reply:
I'm referring to the following comment part:

> or the one it can be masqueraded to

Reply:
Ah OK, with that I was referring to the output of `ip -o route get 10.0.0.1 | grep -oE 'src [^ ]*' | cut -d' ' -f2` (which had been added in https://github.com/cilium/cilium-cli/pull/1870).

Reply:
I just wanted to say that no masquerading happening in that case, and cilium_host's IP addr is chosen by the routing layer.

Reply:
I've rephrased the comment, and removed the reference to the masquerade process.